src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -137,6 +138,33 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
     api_instance = kube_client.CoreV1Api()
     node_api_response = utils.validate_node_api_response(api_instance, None)
 
+    # Pre onboarding checks
+    try:
+        absolute_path = os.path.abspath(os.path.dirname(__file__))
+        kubectl_client_location = install_kubectl_client()
+        helm_client_location = install_helm_client()
+        release_namespace = get_release_namespace(kube_config, kube_context, helm_client_location)
+        diagnostic_checks = "Failed"
+        batchv1_api_instance = kube_client.BatchV1Api()
+        corev1_api_instance = kube_client.CoreV1Api()
+        # Performing pre onboarding inspector container check
+        diagnostic_checks = precheckutils.check_preonboarding_inspector_container(corev1_api_instance, batchv1_api_instance, absolute_path, helm_client_location, kubectl_client_location, release_namespace, kube_config, kube_context, http_proxy, https_proxy, no_proxy, proxy_cert)
+        # If all the checks passed then display no error found
+        all_checks_passed = True
+        # for checks in diagnostic_checks:
+        if diagnostic_checks != consts.Diagnostic_Check_Passed:
+                all_checks_passed = False
+
+    except Exception as e:
+        logger.warning("Exception occured : {}".format(str(e)))
+
+    # Handling the user manual interrupt
+    except KeyboardInterrupt:
+        raise ManualInterrupt('Process terminated externally.')
+
+    if all_checks_passed is False:
+        return
```

Comment:
we should raise proper errors and send telemetry by categorizing this as user fault

Reply:
if there is an error from the prechecks, meaning DNS or outbound failed , user fault is raised in the check_cluster_outbound_connectivity and check_cluster_DNS functions in the utils

Reply:
But we need to return also with a proper message that we are failing the command because prechecks failed

Reply:
Looks like this is still a simple return.